quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -349,16 +352,30 @@ public Builder<T> isEnabled(BooleanSupplier isEnabled) {
         /**
          * Set the validator, overwriting the current one.
          */
-        public Builder<T> validator(BiConsumer<PropertyMapper<T>, ConfigValue> validator) {
-            this.validator = validator;
+        public Builder<T> validator(Consumer<String> validator) {
+            this.validator = (mapper, value) -> mapper.validateExpectedValues(value,
+                    (c, v) -> validator.accept(v));
+            if (!Objects.equals(this.description, this.option.getDescription())) {
+                throw new AssertionError("appendValidateEnabled was already called");
+            }
             return this;
         }
         
         public Builder<T> appendValidator(BiConsumer<PropertyMapper<T>, ConfigValue> validator) {
```

Comment:
Are we actually using somewhere the `PropertyMapper` for validations? Probably not and only the ConfigValue might be sufficient? => IMHO, it should have the same signature as `validator()`.

Reply:
> Are we actually using somewhere the PropertyMapper for validations? 

Yes with appendValidateEnabled. However it is currently only using the option, which could also be obtained from the builder itself.

> IMHO, it should have the same signature as validator().

While that's possible with a tweak based upon the above, it doesn't really offer much simplication based upon current usage, and it will make it more difficult to do anything more advanced later.

Reply:
> While that's possible with a tweak based upon the above, it doesn't really offer much simplication based upon current usage, and it will make it more difficult to do anything more advanced later.

Yep, agree that might be handy for future advanced usage. Like that might be the case even for the `validator()`, right? 

What about providing both signatures with overloading: 
```
validator(Consumer<String> ...)
validator(BiConsumer<PropertyMapper,String> ...)
```
(or `ConfigValue` instead of strings?)

The same for `appendValidator`. I think it will not introduce any additional complexity as it would be very trivial and useful.

WDYT?

Reply:
> (or ConfigValue instead of strings?)

For list item validation that would implie creating ConfigValue wrappers for each item, which I would like to avoid.

> WDYT?

My approach is to start with the minimum and let additional signatures be added as needed. It doesn't bother me that appendValidator has a different signature - as is it's used internally once, and externally once (which doesn't depend on any of the arguments). So it can easily be changed later if desired.

Reply:
> For list item validation that would implie creating ConfigValue wrappers for each item, which I would like to avoid.

+1 

> My approach is to start with the minimum and let additional signatures be added as needed. It doesn't bother me that appendValidator has a different signature - as is it's used internally once, and externally once (which doesn't depend on any of the arguments). So it can easily be changed later if desired.

Got it. I'm just trying to make a good DevEx for the prop mappers and options, as we're not alone who's creating the options and possibly adding the validators, etc.

But the extension of it might be done later as it'd be a straightforward task. 

So, ok.